### PR TITLE
Fix CRI-O installation information

### PR DESCRIPTION
**Pull request by haircommander**
update Ubuntu, CentOS and Debian installation instructions
add Fedora instructions

### CODE REVIEW

**Comment by haircommander** [comment]
PTAL @TomSweeneyRedHat

File: content/en/docs/setup/production-environment/container-runtimes.md [hunk 1]
```diff
@@ -221,71 +221,107 @@ sysctl --system
...
+
+| Operating system | $OS               |
+| ---------------- | ----------------- |
+| Centos 8         | `CentOS_8`        |
+| Centos 8 Stream  | `CentOS_8_Stream` |
+| Centos 7         | `CentOS_7`        |
+<br />
+Then, set `$VERSION` to be the cri-o version matching your kubernetes version.                                                                                                                                                                
+For instance, if you want to install cri-o 1.17, `VERSION=1.17`                                                                                                                                                                               
+We also support pinning to a particular release. To install 1.17.3, `VERSION=1.17:1.17.3`
```
Comment (TomSweeneyRedHat) [inline content/en/docs/setup/production-environment/container-runtimes.md:291]: is 1.17 the latest on CENTOS?  If 1.18 is, I'd bump this example

**Comment by TomSweeneyRedHat** [comment]
One question, otherwise LGTM

**Comment by kbhawkey** [comment]
Page preview:
https://deploy-preview-23000--kubernetes-io-master-staging.netlify.app/docs/setup/production-environment/container-runtimes/

File: content/en/docs/setup/production-environment/container-runtimes.md (hunk 1 above)
Comment (haircommander) [reply content/en/docs/setup/production-environment/container-runtimes.md]: fixed

File: content/en/docs/setup/production-environment/container-runtimes.md [hunk 2]
```diff
@@ -221,71 +221,107 @@ sysctl --system
...
+| Operating system | $OS               |
+| ---------------- | ----------------- |
+| Ubuntu 20.04     | `xUbuntu_20.04`   |
+| Ubuntu 19.10     | `xUbuntu_19.10`   |
+| Ubuntu 19.04     | `xUbuntu_19.04`   |
+| Ubuntu 18.04     | `xUbuntu_18.04`   |
+<br />
+Then, set `$VERSION` to be the cri-o version matching your kubernetes version.
+For instance, if you want to install cri-o 1.18, `VERSION=1.18`.
+We also support pinning to a particular release. To install 1.18.3, `VERSION=1.18:1.18.3`.
```
Comment (kbhawkey) [inline content/en/docs/setup/production-environment/container-runtimes.md:263]: nit: There are several places where you use the phrase, "We also support pinning".
What do you think about rewording the phrase to something like:
CRI-O supports pinning to a specific release.
For example, to install version 1.18.3 set `VERSION=1.18:1.18.3`.
OR
You can pin your installation to a specific release. To install version 1.18.3, set `VERSION=1.18:1.18.3`.

File: content/en/docs/setup/production-environment/container-runtimes.md [hunk 3]
```diff
@@ -221,71 +221,107 @@ sysctl --system
...
 
-{{% tab name="CentOS/RHEL 7.4+" %}}
+{{% tab name="CentOS" %}}
 
-```shell
-# Install prerequisites
-curl -L -o /etc/yum.repos.d/devel:kubic:libcontainers:stable.repo https://download.opensuse.org/repositories/devel:kubic:libcontainers:stable/CentOS_7/devel:kubic:libcontainers:stable.repo
-curl -L -o /etc/yum.repos.d/devel:kubic:libcontainers:stable:cri-o:{{< skew latestVersion >}}.repo https://download.opensuse.org/repositories/devel:kubic:libcontainers:stable:cri-o:{{< skew latestVersion >}}/CentOS_7/devel:kubic:libcontainers:stable:cri-o:{{< skew latestVersion >}}.repo
-```
+To install on the following operating systems, set the environment variable $OS as the appropriate field in the following table:
```
Comment (kbhawkey) [inline content/en/docs/setup/production-environment/container-runtimes.md:281]: nit: To install CRI-O on the following ...

File: content/en/docs/setup/production-environment/container-runtimes.md [hunk 4]
```diff
@@ -221,71 +221,107 @@ sysctl --system
...
+
+| Operating system | $OS               |
+| ---------------- | ----------------- |
+| Centos 8         | `CentOS_8`        |
+| Centos 8 Stream  | `CentOS_8_Stream` |
+| Centos 7         | `CentOS_7`        |
+<br />
```
Comment (kbhawkey) [inline content/en/docs/setup/production-environment/container-runtimes.md:288]: Is the `<br>` required to provide spacing between the table and the following text?

Comment (haircommander) [reply content/en/docs/setup/production-environment/container-runtimes.md:288]: yeah!

**Comment by haircommander** [comment]
addressed as suggested @kbhawkey PTAL

**Comment by inductor** [comment]
@haircommander 

Is this design intended? looks kinda weird to me 🤔 

image

**Comment by haircommander** [comment]
If I'm being honest, the table format is not being rendered super well. I think it is clear enough to be understood, but I'm happy to adopt a change that would look better.

File: content/en/docs/setup/production-environment/container-runtimes.md [hunk 5]
```diff
@@ -221,71 +221,110 @@ sysctl --system
...
-```
+To install on the following operating systems, set the environment variable $OS as the appropriate field in the following table:
 
+| Operating system | $OS               |
+| ---------------- | ----------------- |
+| Ubuntu 20.04     | `xUbuntu_20.04`   |
+| Ubuntu 19.10     | `xUbuntu_19.10`   |
+| Ubuntu 19.04     | `xUbuntu_19.04`   |
+| Ubuntu 18.04     | `xUbuntu_18.04`   |
+<br />
```
Comment (kbhawkey) [inline content/en/docs/setup/production-environment/container-runtimes.md:261]: Hi @haircommander .
I suspect the parser is not happy with the `<br>` elements.
What happens if these are removed? Is there no spacing after the table
and before the next line of text?

Comment (haircommander) [reply content/en/docs/setup/production-environment/container-runtimes.md:261]: yeah the table comes immediately before. It definitely looks weird. I can push an updated version if you'd like

Comment (haircommander) [reply content/en/docs/setup/production-environment/container-runtimes.md]: lmk what you think, I left it as a separate commit to be easily reverted. I think the oddly spaced out table is better

Comment (haircommander) [reply content/en/docs/setup/production-environment/container-runtimes.md:263]: actually, I updated the version, adding a newline between br and the table. that fixed the issue

Comment (kbhawkey) [reply content/en/docs/setup/production-environment/container-runtimes.md:263]: Thanks @haircommander. The tables look good.

File: content/en/docs/setup/production-environment/container-runtimes.md [hunk 6]
```diff
@@ -221,71 +221,113 @@ sysctl --system
...
 
 {{% tab name="openSUSE Tumbleweed" %}}
 
 ```shell
 sudo zypper install cri-o
 ```
+{{% /tab %}}
+{{% tab name="Fedora" %}}
+
+Set `$VERSION` to be the cri-o version matching your kubernetes version.
```
Comment (zacharysarah) [inline content/en/docs/setup/production-environment/container-runtimes.md:317]:
```suggestion
Set `$VERSION` to the CRI-O version that matches your Kubernetes version.                                                                                                                                                                
```

File: content/en/docs/setup/production-environment/container-runtimes.md [hunk 7]
```diff
@@ -221,71 +221,113 @@ sysctl --system
...
 {{% tab name="openSUSE Tumbleweed" %}}
 
 ```shell
 sudo zypper install cri-o
 ```
+{{% /tab %}}
+{{% tab name="Fedora" %}}
+
+Set `$VERSION` to be the cri-o version matching your kubernetes version.                                                                                                                                                                
+For instance, if you want to install cri-o 1.18, `VERSION=1.18`
```
Comment (zacharysarah) [inline content/en/docs/setup/production-environment/container-runtimes.md:318]: Capitalize consistently.
```suggestion
For instance, if you want to install CRI-O 1.18, `VERSION=1.18`                                                                                                                                                                               
```

File: content/en/docs/setup/production-environment/container-runtimes.md [hunk 8]
```diff
@@ -221,71 +221,113 @@ sysctl --system
...
+{{% /tab %}}
+{{% tab name="Fedora" %}}
+
+Set `$VERSION` to be the cri-o version matching your kubernetes version.                                                                                                                                                                
+For instance, if you want to install cri-o 1.18, `VERSION=1.18`                                                                                                                                                                               
+You can find available versions with:
+```shell
+dnf module list cri-o
+```
+We do not support pinning to specific releases on Fedora.
```
Comment (zacharysarah) [inline content/en/docs/setup/production-environment/container-runtimes.md:323]: Avoid using ["we"](https://kubernetes.io/docs/contribute/style/style-guide/#avoid-using-we).
```suggestion
CRI-O does not support pinning to specific releases on Fedora.
```

File: content/en/docs/setup/production-environment/container-runtimes.md [hunk 9]
```diff
@@ -221,71 +221,113 @@ sysctl --system
 {{< tabs name="tab-cri-cri-o-installation" >}}
 {{% tab name="Debian" %}}
 
-```shell
-# Debian Unstable/Sid
-echo 'deb http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/Debian_Unstable/ /' > /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list
-wget -nv https://download.opensuse.org/repositories/devel:kubic:libcontainers:stable/Debian_Unstable/Release.key -O- | sudo apt-key add -
-```
+To install CRI-O on the following operating systems, set the environment variable $OS as the appropriate field in the following table:
```
Comment (zacharysarah) [inline content/en/docs/setup/production-environment/container-runtimes.md:224]: Use code format for [inline code](https://kubernetes.io/docs/contribute/style/style-guide/#use-code-style-for-inline-code-and-commands).
```suggestion
To install CRI-O on the following operating systems, set the environment variable $OS to the appropriate field in the following table:
```

File: content/en/docs/setup/production-environment/container-runtimes.md [hunk 10]
```diff
@@ -221,71 +221,113 @@ sysctl --system
...
+| Debian Unstable  | `Debian_Unstable` |
+| Debian Testing   | `Debian_Testing`  |
 
-```shell
-# Debian 10
-echo 'deb http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/Debian_10/ /' > /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list
-wget -nv https://download.opensuse.org/repositories/devel:kubic:libcontainers:stable/Debian_10/Release.key -O- | sudo apt-key add -
-```
+<br />
+Then, set `$VERSION` to be the cri-o version matching your kubernetes version.
```
Comment (zacharysarah) [inline content/en/docs/setup/production-environment/container-runtimes.md:232]: Capitalize consistently.
```suggestion
Then, set `$VERSION` to be the CRI-O version matching your kubernetes version.
```

File: content/en/docs/setup/production-environment/container-runtimes.md [hunk 11]
```diff
@@ -221,71 +221,113 @@ sysctl --system
...
+For instance, if you want to install cri-o 1.18, set `VERSION=1.18`.
```
Comment (zacharysarah) [inline content/en/docs/setup/production-environment/container-runtimes.md:233]: Capitalize consistently.
```suggestion
For instance, if you want to install CRI-O 1.18, set `VERSION=1.18`.
```

File: content/en/docs/setup/production-environment/container-runtimes.md [hunk 12]
```diff
@@ -221,71 +221,113 @@ sysctl --system
...
+{{% tab name="Ubuntu" %}}
 
-```shell
-# Configure package repository
-. /etc/os-release
-sudo sh -c "echo 'deb http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/x${NAME}_${VERSION_ID}/ /' > /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list"
-wget -nv https://download.opensuse.org/repositories/devel:kubic:libcontainers:stable/x${NAME}_${VERSION_ID}/Release.key -O- | sudo apt-key add -
-sudo apt-get update
-```
+To install on the following operating systems, set the environment variable $OS as the appropriate field in the following table:
```
Comment (zacharysarah) [inline content/en/docs/setup/production-environment/container-runtimes.md:254]:
```suggestion
To install CRI-O on the following operating systems, set the environment variable $OS as the appropriate field in the following table:
```

File: content/en/docs/setup/production-environment/container-runtimes.md [hunk 13]
```diff
@@ -221,71 +221,113 @@ sysctl --system
...
 
+| Operating system | $OS               |
+| ---------------- | ----------------- |
+| Ubuntu 20.04     | `xUbuntu_20.04`   |
+| Ubuntu 19.10     | `xUbuntu_19.10`   |
+| Ubuntu 19.04     | `xUbuntu_19.04`   |
+| Ubuntu 18.04     | `xUbuntu_18.04`   |
+
+<br />
+Then, set `$VERSION` to be the cri-o version matching your kubernetes version.
```
Comment (zacharysarah) [inline content/en/docs/setup/production-environment/container-runtimes.md:264]:
```suggestion
Then, set `$VERSION` to the CRI-O version that matches your Kubernetes version.                                                                                                                                                                
```

File: content/en/docs/setup/production-environment/container-runtimes.md [hunk 14]
```diff
@@ -221,71 +221,113 @@ sysctl --system
...
+For instance, if you want to install cri-o 1.18, set `VERSION=1.18`.
```
Comment (zacharysarah) [inline content/en/docs/setup/production-environment/container-runtimes.md:265]: Capitalize consistently.
```suggestion
For instance, if you want to install CRI-O 1.18, set `VERSION=1.18`.
```

File: content/en/docs/setup/production-environment/container-runtimes.md [hunk 15]
```diff
@@ -221,71 +221,113 @@ sysctl --system
...
 
-{{% tab name="CentOS/RHEL 7.4+" %}}
+{{% tab name="CentOS" %}}
 
-```shell
-# Install prerequisites
-curl -L -o /etc/yum.repos.d/devel:kubic:libcontainers:stable.repo https://download.opensuse.org/repositories/devel:kubic:libcontainers:stable/CentOS_7/devel:kubic:libcontainers:stable.repo
-curl -L -o /etc/yum.repos.d/devel:kubic:libcontainers:stable:cri-o:{{< skew latestVersion >}}.repo https://download.opensuse.org/repositories/devel:kubic:libcontainers:stable:cri-o:{{< skew latestVersion >}}/CentOS_7/devel:kubic:libcontainers:stable:cri-o:{{< skew latestVersion >}}.repo
-```
+To install on the following operating systems, set the environment variable $OS as the appropriate field in the following table:
```
Comment (zacharysarah) [inline content/en/docs/setup/production-environment/container-runtimes.md:285]:
```suggestion
To install CRI- O on the following operating systems, set the environment variable $OS as the appropriate field in the following table:
```

File: content/en/docs/setup/production-environment/container-runtimes.md [hunk 16]
```diff
@@ -221,71 +221,113 @@ sysctl --system
...
 
+| Operating system | $OS               |
+| ---------------- | ----------------- |
+| Centos 8         | `CentOS_8`        |
+| Centos 8 Stream  | `CentOS_8_Stream` |
+| Centos 7         | `CentOS_7`        |
+
+<br />
+Then, set `$VERSION` to be the cri-o version matching your kubernetes version.
```
Comment (zacharysarah) [inline content/en/docs/setup/production-environment/container-runtimes.md:294]: Capitalize consistently.
```suggestion
Then, set `$VERSION` to the CRI-O version that matches your Kubernetes version.
```

File: content/en/docs/setup/production-environment/container-runtimes.md [hunk 17]
```diff
@@ -221,71 +221,113 @@ sysctl --system
...
+For instance, if you want to install cri-o 1.18, set `VERSION=1.18`.
```
Comment (zacharysarah) [inline content/en/docs/setup/production-environment/container-runtimes.md:295]:
```suggestion
For instance, if you want to install CRI-O 1.18, set `VERSION=1.18`.
```

**Comment by haircommander** [comment]
thank you @zacharysarah, updated as suggested

**Pull request merged**